### PR TITLE
optimize humanoid demo

### PR DESCRIPTION
**Pull request by PaulAroo**

### CODE REVIEW

File: s/demo.ts [hunk 1]
```diff
@@ -87,67 +95,125 @@ void async function main() {
...
+
+		case "medium":
+			// {
+			// 	shadow_generator.autoCalcDepthBounds = true
+			// 	shadow_generator.bias = 0.01
+			// 	shadow_generator.normalBias = 0
+			// }
+
+			{
+				const pipeline = new DefaultRenderingPipeline("default", false, scene, [camera])
```
Comment (PaulAroo) [inline s/demo.ts:123]: @chase-moskal I found out, setting the `HDR` texture to false helps with the performance, but I still can't get a good `fps` with either image processing, `ao` or shadow tweaks on... 🤔

File: s/demo.ts [hunk 2]
```diff
@@ -87,67 +95,125 @@ void async function main() {
...
+				// pipeline.imageProcessing.toneMappingEnabled = true
+				// pipeline.imageProcessing.toneMappingType = TonemappingOperator.Photographic
+
+				// pipeline.grainEnabled = true
+				// pipeline.grain.intensity = 5
+				// pipeline.grain.animated = true
+			}
+
+			// {
+			// 	const ao_settings = 0.75
```
Comment (PaulAroo) [inline s/demo.ts:152]: I left these commented out cause it's still a work on progress, hoping to delete the unused ones when it has finally been decided which ones stay

File: s/demo.ts (hunk 1 above)
Comment (chase-moskal) [reply s/demo.ts:123]: for medium quality, we really want ao+shadows+bloom enabled, albeit configured frugally

but not HDR or much else that might harm performance

Comment (PaulAroo) [reply s/demo.ts:123]: okay, I'll add one more commit that enables ao but with low configurations

File: s/demo.ts [hunk 3]
```diff
@@ -148,21 +148,22 @@ void async function main() {
...
+				ao.base = 0.15
+				ao.maxZ = 600
+				ao.samples = 4
+				ao.minZAspect = 0.5
+				// ao.expensiveBlur = false;
+			}
 			break
 
 		case "high":
+			theater.settings.resolutionScale = 150
```
Comment (PaulAroo) [inline s/demo.ts:166]: what do you think of this @chase-moskal and @zenkyuv , since it's high quality mode it is okay to start at a much higher resolution

Comment (zenkyuv) [reply s/demo.ts:166]: I think if we already change resolution for other modes then that would be reasonable, i think either we set 100% resolution for every mode or either we do like now, appropiate to mode by setting ideal cost effective resolution for it

Comment (chase-moskal) [reply s/demo.ts:166]: in most cases, a resolution higher than 100 is a complete waste that is catastrophic to the framerate. "overclocking" the resolution like that is only useful for users with higher pixel density displays where the device pixel ratio detection in babylon is malfunctioning, which will be quite rare.

most gamers, rightly so, will have the expectation that the game will render at 100% resolution.

@lonnie-ralfs and i were also discussing this, and we feel strongly that default 100% is preferable for all modes, only with the exception of "low" at 50% being a drastic measure of last resort to attempt playability for low-end phones.

Comment (PaulAroo) [reply s/demo.ts:166]: Okay

Comment (zenkyuv) [reply s/demo.ts]: Yes i think that would be ideal, i have never seen any game that have resolution set higher than your monitor resolution on higher settings

**Pull request merged**